### ISSUE TITLE
Unified interface for Cookies / cookie.

### ISSUE THREAD

**Issue by redstrike** [js-cookie/js-cookie#125]
Coming from PHP world and conceived the usefulness of PSR ([PHP Standards Recommendations](http://www.php-fig.org/psr/)). I think that we should have a unified interface for a stable and lightweight component like Cookie.

Last night, it cost me a lot of time to compare the different of Cookie libraries like [js-cookie](https://github.com/js-cookie/js-cookie) vs [ScottHamper/Cookies](https://github.com/ScottHamper/Cookies) vs [florian/cookie.js](https://github.com/florian/cookie.js). I wonder why libraries' authors don't spend time together to design a unified interface for Cookie. We also don't have the feature that big company like Google used to set "multiple values to a key" (Youtube.PREF: {hl=en&abc=xyz&...}).

An interoperable unified interface helps library user much when choosing a good one and swapping to the right one in the forest of libraries of JS community.

**Comment by FagnerMartinsBrack** [comment]
I will interpret your concerns as:

> It is hard to discover among a lot of libraries, a robust and maintained interface that solves all parts of a problem given a standard use-case

The web is not owned by one person, but by an organization. The only way to get a standard interface for something that will work above everything else is passing through a recommendation to the specific standards body. I really don't think that they would consider a new cookie interface because that would need to be frozen forever and the DOM already have a "cookie" API (even though it's crap).

In js-cookie and similar libs it is possible to create a major version bump that make some relevant changes to the API because all of them (hopefully) follows semver.org. Such kind of bump was already made from jquery-cookie to js-cookie 2. This kind of thing cannot be done if browsers choose to implement it, for example.

So, browser implementation is out of the equation. Regarding lib authors standardization I don't see a point in creating a common interface because each project has it's own opinionated interfaces. Creating a common interface would make obsolete the need for multiple packages.

We created js-cookie organization to make it simple and standard the handling of something as trivial as a cookie, using some knowledge we had from maintaining jquery cookie to create an awesome interface for general use cases. I learned a lot with the [version 2 rewrite in PR #5](https://github.com/js-cookie/js-cookie/pull/5), including how RFC 6265 works from top down. Maybe other lib authors do not have the time to dedicate so much for this sort of things and therefore that might explain why each project has a different interface, features or behavior.

Unfortunately npm is too granular due to its zero barrier of entry and the JS ecossystem is big. I don't see a solution in the near future for reducing the barrier of discovery for relevant libs and plugins out there unless someone that is popular enough step up and create some sort of "curated" list of libs, plugins, etc. In the meantime, the "curated" list of libs, plugins, etc. are the ones with more stars/watches/followers/whatever.

I would say: construct your heuristics by experience. Because even if the repository [have almost 30k of stars](https://github.com/impress/impress.js) that's not even a guarantee that it is being maintained or even worth using for all purposes.

TL;DR: There is no solution for this problem, at least not right now.

**Comment by redstrike** [comment]
Thank you for spending time to reply my opinion. After re-thinking about this problem, I see at least we already have 2 unified methods for cookie (get, set). It's good for most of common project. The main different is other advanced method like getJSON(). Therefore, we could end this issue here, but to be honest, I hope you and other authors can spend a little time to discuss about this idea.
